### DISCUSSION TITLE
Axis Q6075 Raw sensor output

### DISCUSSION THREAD

**Discussion by DEHiggin123** [AxisCommunications/discussions#163] [General] [answered]
Hello,
I have a Q6075-e camera and I noticed that there is a few post processing applications on it. For instance lens distortion correction, for a project I am working on I need to see as much of the screen as possible and would like to run lens distortion correction outside of the camera. How do I do I get the raw camera sensor output? I have already checked Vapix many times and haven't found what I'm looking for.

**Comment by janssen70** [comment]
You probably want a 8-bit RGB or YUV bitmap, not raw. But in either case, that makes 6MByte * 30 fps ~1.44Gbit/second pixel data. 
It's better to work with a stream and decode it
The bitmap.cgi, documented in video API, can give you a single RGB bitmap

**Reply by DEHiggin123** [reply]
axis-cgi/bitmap/image.bmp
This responds with a image with lens distortion correction...

**Reply by DEHiggin123** [reply]
![remove-gopro-fisheye-distortion](https://user-images.githubusercontent.com/115744909/214455968-39b02e3e-2294-4713-8748-e7362617a63d.jpg)
This is what I need.

**Comment by daniel-falk** [comment]
@DEHiggin123 Do you see any fisheye view under the `Video->Streams` tab in the camera? If you find it in the drop-down list there, then you can stream that view using the `?camera=N` parameter in the URL where `N` is the index in the list. See example in this image where I stream the Quad View using RTSP.
![streams](https://user-images.githubusercontent.com/23218783/215028642-114ae897-43af-48e2-8a21-64e025ee4367.png)

**Reply by DEHiggin123** [reply]
/axis-cgi/bitmap/image.bmp?camera=1 : original image
/axis-cgi/bitmap/image.bmp?camera=2 : 400 Bad Request
There is only one image viewable, I am not on a panoramic camera. I am on a Q-6075-e

**Reply by DEHiggin123** [reply]
That is what I need though, I must have a different settings menu than you. And no there is no fisheye option in any settings.

**Reply by daniel-falk** [reply]
Sounds like one of three issues:

- The view modes are disabled
- You might need to specify a resolution
- If the camera rotation is not set, you might need to do that

In the plain config page, make sure that the overview is enabled (I0):
<img width="1512" alt="Screenshot 2023-01-28 at 08 03 09" src="https://user-images.githubusercontent.com/23218783/215252182-59f51fb1-5f0d-4cbd-9f89-177c692dc8c2.png">


You can also check using VAPIX:

First check if the non-dewarped view is enabled:
```
http://<ip>/axis-cgi/param.cgi?action=list&group=Image.I0.Enabled
```

List the supported resolutions:
```
http://127.0.0.1:8080/axis-cgi/param.cgi?action=list&group=Properties.Image.I0.Resolution
```

[Also find the documentation here.](https://www.axis.com/vapix-library/subjects/t10175981/section/t10011928/display)

Unfortunately I do not have any Q-6075-e so I don't know if it looks exactly the same for you (might vary depending on FW version too).

**Reply by DEHiggin123** [reply]
I don't think you understand, the camera is non panoramic. For Image.I0.Enabled it outputs true, for group=Properties.Image.I0.Resolution it cannot find it. But its 1920x1080, I can check via Image.I0.Appearance.Resolution. Now according to the action=list sheet there is only 1 view. This means that the camera most likely does not support dewarped views. So I am going to need a way to get it, is it physically preprocessed then sent to another board?

**Reply by DEHiggin123** [reply]
Properties.Image.Dewarp=
does not even exist in action=list...

**Reply by daniel-falk** [reply, marked as answer]
OK, I understand. In that case I don't think your camera supports view modes, which means that there is a barrel distortion correction parameter affecting all streams globally. A camera with multiple view modes have multiple paths through the image processing pipeline (implemented in the silicon) which allows you to preprocess different view modes differently in parallel and then select which view mode  you stream. Other cameras does only have one path which means that if you change any parameter affecting the IPP (like the BDC does), it will affect all streams regardless of the `camera=..` parameter. This is at least as I understand it.

I guess that you should have a BDC parameter like this one? Changing that should change the rtsp streams too.
![bdc](https://user-images.githubusercontent.com/23218783/215684351-27fc319e-2dbf-4379-bf39-416050ef892d.png)
